### PR TITLE
added first pass of snekspec game state creation

### PR DESCRIPTION
**Pull request by mike-anderson**
it doesn't have to be called snekspec, I just liked 'snekspec by echosec' as a title

There are likely still edge cases that can be found using the test suite, please suggest them.

Also I haven't coded anything for like a year, so please let me know what could be changed coding standard wise. There's a mix of for and foeach that I'm not super fond of.

### CODE REVIEW

File: util/snekspec.ts [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+  const dy = Math.abs(a.y - b.y);
+  return dx + dy === 1;
+}
+
+/**
+ * Compute the path from head to tail from the scanned collection
+ * of snake body coordinates
+ * @param snake
+ */
+function untagleSnake(snake: Array<ICoordinate>): Array<ICoordinate> {
```
Comment (samueljohnwilson) [inline util/snekspec.ts]: Just noticed a typo here

**Comment by mike-anderson** [comment]
had a suspicion the snake generator didn't quite work for snakes that looped back on themselves.
Turns out it didn't but it was a pretty easy fix - so far I am a huge fan of jest.

Also, this is not an efficient process - a flood fill snake that takes up 3 lines of the board takes 20 seconds to solve on my machine. This test is edited out to save time on CI

I've changed the jest config to only run files that match either `__tests__/*.test.ts` or `/test.ts` so that I could include supporting files in the test directory that shouldn't be part of code coverage such as mocks. Let me know if this is ok.

**Review by dmarcantonio — APPROVED** [review]
I barely looked at this!

**Review by brandonb927 — APPROVED** [review]
